### PR TITLE
Transforms to new supported models style

### PR DESCRIPTION
**Pull request by martinju**
Updates docs, vignette and readme. Joss paper handled separately.

### CODE REVIEW

File: R/models.R [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
 #' @param newdata A data frame (or matrix) in which to look for variables with which to predict.
 #'
 #' @details The following models are currently supported:
 #' \itemize{
-#' \item \code{\link[stats]{lm}}
-#' \item \code{\link[stats]{glm}}
-#' \item \code{\link[ranger]{ranger}}
-#' \item \code{\link[mgcv]{mgcv}}
-#' \item \code{\link[xgboost]{xgboost}}
+#' \item \code{\link[stats:lm]{stats::lm}}
```
Comment (nikolase90) [inline R/models.R]: Why this change?

File: README.md [hunk 2]
```diff
@@ -157,6 +157,11 @@ print(explanation$dt)
 
 # Finally we plot the resulting explanations
 plot(explanation)
+#> Registered S3 methods overwritten by 'ggplot2':
```
Comment (nikolase90) [inline README.md]: Regenerate the file and see if this change gets removed? I tested locally and it was removed. Let me know if I should push that.

File: R/models.R (hunk 1 above)
Comment (martinju) [reply R/models.R]: Maybe not needed for lm and glm, but this is more specific and in accordance with other references to supported models in the package. In the brief description we are more "sloppy" to simplify the read.

The link text is now stats::lm instead of just lm etc, while the link is the same.

File: README.md (hunk 2 above)
Comment (martinju) [reply README.md]: Fixed in 14a9f67

File: R/models.R (hunk 1 above)
Comment (nikolase90) [reply R/models.R]: Oki. As long as the links still work, ref
> There are two tags that make it easier for people to navigate between help files
...
\code{\link[packagename]{functioname}}.
If you have a family of related functions where every function should link to every other ...

http://r-pkgs.had.co.nz/man.html

Comment (martinju) [reply R/models.R]: Yes, it works. See further down in that link...

\code{\link[MASS:abbey]{name}}: link to function in another package, but show name.

Comment (nikolase90) [reply R/models.R]: Awesome!
